runtime-modules/bounty/src/lib.rs

```diff
@@ -1052,64 +1310,65 @@ decl_module! {
 
             let entry = Self::ensure_work_entry_exists(&entry_id)?;
 
+            Self::ensure_work_entry_ownership(&entry, &member_id)?;
+
             //
             // == MUTATION SAFE ==
             //
 
-            Self::unlock_work_entry_stake_with_possible_penalty(&bounty, &entry);
-
-            Self::remove_work_entry(&bounty_id, &entry_id);
+            // Update entry
+            <Entries<T>>::mutate(entry_id, |entry| {
+                entry.work_submitted = true;
+            });
 
-            Self::deposit_event(RawEvent::WorkEntryWithdrawn(bounty_id, entry_id, member_id));
+            Self::deposit_event(RawEvent::WorkSubmitted(bounty_id, entry_id, member_id, work_data));
         }
 
-        /// Submit work for a bounty.
+        /// end bounty working period.
         /// # <weight>
         ///
         /// ## weight
-        /// `O (N)`
-        /// - `N` is the work_data length,
+        /// `O (1)`
         /// - db:
         ///    - `O(1)` doesn't depend on the state or parameters
         /// # </weight>
-        #[weight =  WeightInfoBounty::<T>::submit_work(work_data.len().saturated_into())]
-        pub fn submit_work(
+        #[weight = WeightInfoBounty::<T>::end_working_period()]
+        pub fn end_working_period(
             origin,
-            member_id: MemberId<T>,
             bounty_id: T::BountyId,
-            entry_id: T::EntryId,
-            work_data: Vec<u8>
         ) {
-            T::Membership::ensure_member_controller_account_origin(origin, member_id)?;
 
             let bounty = Self::ensure_bounty_exists(&bounty_id)?;
+            let current_oracle = bounty.creation_params.oracle.clone();
+            let oracle_reward = bounty.creation_params.oracle_reward;
+            //Checks if the function caller (origin) is current oracle
+            BountyActorManager::<T>::ensure_bounty_actor_manager(
+                origin,
+                current_oracle.clone(),
+            )?;
 
             let current_bounty_stage = Self::get_bounty_stage(&bounty);
 
             Self::ensure_bounty_stage(current_bounty_stage, BountyStage::WorkSubmission)?;
-
-            Self::ensure_work_entry_exists(&entry_id)?;
-
             //
             // == MUTATION SAFE ==
             //
+            if bounty.active_work_entry_count == 0{
```

Comment:
1. This does solvency check, which we should not.
2. I think its better to try to limit the number of places payouts are happening, as it becomes increasingly complex to keep track of whether it has already happened or not, lets just allow this to be done in the withdrawal stage as usual.

Reply:
If you want all payouts (oracle reward and cherry) to be placed in the withdrawal, I have to modify the bountymilestone to include flags indicating which path the bounty followed, in some situations, a failed withdrawal can mean different payouts to different actors (ex creator or oracle).